fairing/utils.py

```diff
@@ -33,3 +35,51 @@ def crc(file_name):
 
 def random_tag():
     return str(uuid.uuid4()).split('-')[0]
+
+
+def guess_preprocessor(entry_point, *args, **kwargs):
+    #Trying to avoid any clyclic dependencies by importing locally within the method
+    from fairing.functions.function_shim import get_execution_obj_type, ObjectType
+    from fairing.preprocessors.function import FunctionPreProcessor
+    
+    if get_execution_obj_type(entry_point) != ObjectType.NOT_SUPPORTED:
+        return FunctionPreProcessor(entry_point, *args, **kwargs)
+    else:
+        raise NotImplementedError(
+            "obj param should be a function or a class, got {}".format(type(entry_point)))
+
+
+def guess_docker_registry():
```

Comment:
these checks belong in the actual functions that need them if that makes sense

Reply:
Removed guess_docker_registry and made docker registry required param for now. Let's address how to provide good defaults based on the environment (KF on-prem, KF on GKE, managed backend, etc)